### PR TITLE
[FIX] barcode: Update the product quantities on scanning product barcode

### PR DESCRIPTION
**Pull request by abh-odoo**
Task: https://www.odoo.com/web#id=746620&view_type=form&model=project.issue&action=963&active_id=681&menu_id=4720

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/js/views/basic/basic_controller.js [hunk 1]
```diff
@@ -114,6 +114,14 @@ var BasicController = AbstractController.extend(FieldManagerMixin, {
         return this.mutex.exec(function () {})
             .then(this._discardChanges.bind(this, recordID || this.handle, options));
     },
+    _nextPage: function () {
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/basic/basic_controller.js]: those functions are private, they should be in the Private section, ordered alphabetically and properly documented. See our new (saas-16) coding guidelines: https://github.com/odoo/odoo/wiki/Javascript-coding-guidelines

File: addons/web/static/src/js/views/form/form_controller.js [hunk 2]
```diff
@@ -70,6 +70,18 @@ var FormController = BasicController.extend({
...
+    },
+    nextPage:function() {
+        return this._nextPage();
+    },
+    previousPage:function() {
+        return this._previousPage();
+    },
+    discardChange: function() {
+        return this.discardChanges();
+    },
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/form/form_controller.js]: i think that those functions are not necessary. You could directly call the right function in your include in barcodes (a private function can't be called by the outside world, but in this case you are still in the form_controller.
For toEditMode, you can define an inline function as you already did for 'O-CMD.SAVE'

Comment (vas-odoo) [inline addons/web/static/src/js/views/form/form_controller.js]: I have done the changes as you have suggested.
Thank you..

**Comment by aab-odoo** [comment]
Just merged in saas-16. Thanks

**Pull request closed without merging**